python/cuml/ensemble/randomforest.pyx

```diff
@@ -0,0 +1,306 @@
+import ctypes
+import cudf
+import numpy as np
+import warnings
+import cupy
+from cuml.utils import get_cudf_column_ptr, get_dev_array_ptr, \
+    input_to_dev_array, zeros
+from numba import cuda
+from libcpp cimport bool
+from libc.stdint cimport uintptr_t
+from libc.stdlib cimport calloc, malloc, free
+from cuml.common.base import Base
+from cuml.common.handle cimport cumlHandle
+cimport cuml.common.handle
+cimport cuml.common.cuda
+
+cdef extern from "randomforest/randomforest.h" namespace "ML":
+
+    cdef struct RF_metrics:
+        float accuracy
+
+    cdef enum RF_type:
+        CLASSIFICATION,
+        REGRESSION
+
+    cdef struct RF_params:
+        pass
+
+    cdef cppclass rf:
+        pass
+
+    cdef cppclass rfClassifier[T]:
+        rfClassifier(RF_params) except +
+
+    cdef void fit(cumlHandle & handle,
+                  rfClassifier[float] *,
+                  float *,
+                  int,
+                  int,
+                  int *,
+                  int)
```

Comment:
Should add  `except +` to the end of each fit and predict function to that C++ exceptions are passed through. (E.g. in the case of running out of shared memory)